source/core/CEntryPoints.cpp

```diff
@@ -335,6 +335,33 @@ VIREO_EXPORT EggShellResult Data_GetArrayMetadata(TypeManagerRef tm,
     return kEggShellResult_Success;
 }
 //------------------------------------------------------------
+//! Get the starting location of the first element of an Array / String type in memory
+// This function returns the start address of where elements would appear in memory (returns address even if length zero)
+VIREO_EXPORT void* Data_GetArrayBegin(const void* pData)
+{
+    TypedArrayCoreRef arrayObject = *(TypedArrayCoreRef*)pData;
+    VIREO_ASSERT(TypedArrayCore::ValidateHandle(arrayObject));
+    return arrayObject->BeginAt(0);
+}
+//------------------------------------------------------------
+//! Get the values for dimensions of the array. Assumes dimensions target is of length equal to rank
```

Comment:
Who is responsible for the size of the dimensions array? Please add a comment related to that. This function has the potential to write beyond the size of dimensions if the size is not properly taken care of. Consider passing dimensions length parameter if that makes sense.

Reply:
If you pass the dimensions length, it can be used to validate proper array bounds.

Reply:
That's what the "Assumes dimensions target is of length equal to rank" is trying to convey. Is it clearer to say "Caller must allocate an array of length rank for the dimensions"? What's the best way to convey that the caller is responsible?